cli/src/template/ios/MoproApp/ContentView.swift

```diff
@@ -196,5 +203,102 @@ extension ContentView {
+            return
+        }
+        
+        // Load and parse JSON
+        do {
+            let jsonData = try Data(contentsOf: URL(fileURLWithPath: inputJsonPath))
+            let decoder = JSONDecoder()
+            let inputData = try decoder.decode(ZkEmailInputTest.self, from: jsonData)
+            
+            // Convert to the format expected by proveZkemail
+            let inputs: [String] =
+                inputData.header.storage.map { String($0) } +
+                [String(inputData.header.len)] +
+                inputData.pubkey.modulus +
+                inputData.pubkey.redc +
+                inputData.signature +
+                [String(inputData.date_index)] +
+                [String(inputData.subject_sequence.index)] +
+                [String(inputData.subject_sequence.length)] +
+                [String(inputData.from_header_sequence.index)] +
+                [String(inputData.from_header_sequence.length)] +
+                [String(inputData.from_address_sequence.index)] +
+                [String(inputData.from_address_sequence.length)]
+            
+            
+            // Run in background thread
+            DispatchQueue.global(qos: .userInitiated).async {
+                let start = CFAbsoluteTimeGetCurrent()
+                do {
+                    // Generate the proof
+                    let proofData = try! generateNoirProof(circuitPath: zkemailCircuitPath, srsPath: zkemailSrsPath, inputs: inputs)
+                    
+                    let end = CFAbsoluteTimeGetCurrent()
+                    let timeTaken = end - start
+                    
+                    generatedNoirProof = proofData
+                    textViewText += "\(String(format: "%.3f", timeTaken))s 1️⃣\n"
+                    
+                    isNoirVerifyButtonEnabled = true
+                } catch {
+                    textViewText += "Proof generation failed: \(error.localizedDescription)\n"
+                }
+            }
+            
+        } catch {
+            textViewText += "Error loading or parsing input JSON: \(error.localizedDescription)\n"
+        }
+    }
+
+    func runNoirVerifyAction() {
+        guard let proofData = generatedNoirProof else {
+            textViewText += "Error: Proof data is not available. Generate proof first.\n"
+            return
+        }
+
+        textViewText += "Verifying zkEmail proof...\n"
+        
+        // Get the path to the SRS file in the app bundle
+        guard let srsPath = Bundle.main.path(forResource: "srs", ofType: "local") else {
+            textViewText += "Error: Could not find SRS file in app bundle\n"
+            return
+        }
+
+        DispatchQueue.global(qos: .userInitiated).async {
+            let start = CFAbsoluteTimeGetCurrent()
+            do {
+                // Verify the proof
+                let isValid = try! verify_noir_proof(circuitPath: zkemailCircuitPath, proof: proofData)
```

Comment:
should be `verifyNoirProof`

Reply:
fixed in [323ba30](https://github.com/zkmopro/mopro/pull/399/commits/323ba30633ccdf4ce6c601e06607de65b7ebb466)